flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/MRSplitsGetter.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connectors.hive;
+
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.hive.metastore.api.StorageDescriptor;
+import org.apache.hadoop.mapred.InputFormat;
+import org.apache.hadoop.mapred.InputSplit;
+import org.apache.hadoop.mapred.JobConf;
+import org.apache.hadoop.util.ReflectionUtils;
+import org.apache.hadoop.util.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+import static org.apache.flink.util.concurrent.Executors.newDirectExecutorService;
+import static org.apache.hadoop.mapreduce.lib.input.FileInputFormat.INPUT_DIR;
+
+/** Create MR splits by multi thread for hive partitions. */
+public class MRSplitsGetter implements Closeable {
+    private static final Logger LOG = LoggerFactory.getLogger(MRSplitsGetter.class);
+
+    private final ExecutorService executorService;
+
+    public MRSplitsGetter(int threadNum) {
+        LOG.info("Open {} thread to create hive partition splits.", threadNum);
+
+        if (threadNum > 1) {
```

Comment:
Maybe it's better the check threadNum when create hive sourece.
And I think it may be better throw exception when threadNum <=0 instead of assuming it represents single thread.

Reply:
Another choice is to specify that the thread will be set to 1 if the configured value <= 1 in the doc and log it in the method. What do you think?

Reply:
I still prefer to do validation. The behavior should be consistent with the other configurations like `TABLE_EXEC_INFER_SOURCE_PARALLELISM_MAX`, `TABLE_EXEC_INFER_SOURCE_PARALLELISM_MIN`, etc.